lib/Cucumber.j

```diff
@@ -881,103 +894,132 @@ function dumpGuiObject(obj)
+        mouseType = params[1],
         window = [CPApp keyWindow];
 
-    CPLog.debug("Cucapp is about to simulate a right click on the point : " + point);
+    if (mouseType != CPLeftMouseDown && mouseType != CPRightMouseDown)
+        return '{"result" : "INVALID MOUSE TYPE"}';
 
-    [self _perfomMouseEventOnPoint:point toPoint:nil window:window eventType:CPRightMouseDown numberOfClick:1 modifierFlags:params[0]];
+    CPLog.debug("Cucapp is about to simulate a mouse down to the point " + locationWindowPoint);
+
+    [self _dispatchMouseEventWithType:mouseType location:locationWindowPoint modifierFlags:flags clickCount:0 window:window];
+
+    return '{"result" : "OK"}';
 }
 
-- (CPString)simulateRightClick:(CPArray)params
+- (void)simulateMouseDown:(CPArray)params
 {
     var obj = cucumber_objects[params.shift()],
-        locationWindowPoint;
+        locationWindowPoint,
+        flags = params[0],
+        mouseType = params[1],
+        window = [CPApp keyWindow];
 
     if (!obj)
         return '{"result" : "OBJECT NOT FOUND"}';
 
+    if (mouseType != CPLeftMouseDown && mouseType != CPRightMouseDown)
+        return '{"result" : "INVALID MOUSE TYPE"}';
+
     if ([obj superview])
         locationWindowPoint = [[obj superview] convertPointToBase:CGPointMake(CGRectGetMidX([obj frame]), CGRectGetMidY([obj frame]))];
     else
         locationWindowPoint = CGPointMake(CGRectGetMidX([obj frame]), CGRectGetMidY([obj frame]));
 
-    CPLog.debug("Cucapp is about to simulate a right click on the view : " + obj);
+    CPLog.debug("Cucapp is about to simulate a mouse down on the view : " + obj);
 
-    [self _perfomMouseEventOnPoint:locationWindowPoint toPoint:nil window:[obj window] eventType:CPRightMouseDown numberOfClick:1 modifierFlags:params[0]];
+    [self _dispatchMouseEventWithType:mouseType location:locationWindowPoint modifierFlags:flags clickCount:0 window:window];
 
     return '{"result" : "OK"}';
 }
 
-- (void)simulateDoubleClickOnPoint:(CPArray)params
+- (void)simulateMouseUpOnPoint:(CPArray)params
 {
-    var point = CGPointMake(params.shift(), params.shift()),
+    var locationWindowPoint = CGPointMake(params.shift(), params.shift()),
+        flags = params[0],
+        mouseType = params[1],
         window = [CPApp keyWindow];
 
-    CPLog.debug("Cucapp is about to simulate a double click on the point : " + point);
+    if (mouseType != CPLeftMouseUp && mouseType != CPRightMouseUp)
+        return '{"result" : "INVALID MOUSE TYPE"}';
+
+    CPLog.debug("Cucapp is about to simulate a mouse up to the point " + locationWindowPoint);
+
+    [self _dispatchMouseEventWithType:mouseType location:locationWindowPoint modifierFlags:flags clickCount:0 window:window];
 
-    [self _perfomMouseEventOnPoint:point toPoint:nil window:window eventType:CPLeftMouseDown numberOfClick:2 modifierFlags:params[0]];
+    return '{"result" : "OK"}';
 }
 
-- (CPString)simulateDoubleClick:(CPArray)params
+- (void)simulateMouseUp:(CPArray)params
```

Comment:
I'm not sure it's a good idea to use the `simulate*OnPoint` in the `simulate*` methods, because there will double check and double log (Cucapp is about to simulate a mouse down on the view THEN Cucapp is about to simulate a mouse down to the point) and we can't simplify these methods because they are used independently in encumber.
Or maybe we can keep only the `simulate*OnPoint` and drop the others if you want to use the `simulate_*_on_point` in the `simulate_*` methods.
But the behaviour is different between the 2 kind of methods, and we never won't get an error if the view doesn't exist because in any case we'll use `simulate*OnPoint` methods. We have to find another way to do that.